packages/coding-style/src/Rector/ClassMethod/AnnotateThrowablesRector.php

```diff
@@ -0,0 +1,219 @@
+{
+    /**
+     * This is a comment.
+     *
+     * @param int $code
+     */
+    public function throwException(int $code)
+    {
+        throw new \RuntimeException('', $code);
+    }
+}
+PHP
+                    ,
+                    // code after
+                    <<<'PHP'
+class RootExceptionInMethodWithDocblock
+{
+    /**
+     * This is a comment.
+     *
+     * @param int $code
+     * @throws \RuntimeException
+     */
+    public function throwException(int $code)
+    {
+        throw new \RuntimeException('', $code);
+    }
+}
+PHP
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param Throw_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->isThrowableAnnotated($node)) {
+            return null;
+        }
+
+        $this->annotateThrowable($node);
+
+        return $node;
+    }
+
+    private function isThrowableAnnotated(Throw_ $node): bool
+    {
+        $identifiedThrownThrowables = $this->identifyThrownThrowables($node);
+        $phpDocInfo = $this->getThrowingStmtDocblock($node);
+        $alreadyAnnotatedThrowTags = $phpDocInfo->getTagsByName('throws');
+
+        if (empty($alreadyAnnotatedThrowTags)) {
+            return false;
+        }
+
+        $alreadyAnnotatedThrowClassNames = array_map(static function($alreadyAnnotatedThrowTag) {
+            return $alreadyAnnotatedThrowTag->value->type->name;
+        }, $alreadyAnnotatedThrowTags);
+
+        $notAnnotatedThrowables = [];
+        foreach ($identifiedThrownThrowables as $identifiedThrownThrowable) {
+            foreach ($alreadyAnnotatedThrowClassNames as $alreadyAnnotatedThrowClassName) {
+                if ($alreadyAnnotatedThrowClassName === $identifiedThrownThrowable) {
+                    break;
+                }
+
+                if (
+                    ! Strings::contains($alreadyAnnotatedThrowClassName, '\\') &&
+                    Strings::contains($identifiedThrownThrowable, $alreadyAnnotatedThrowClassName) &&
+                    $this->isThrowableImported($node)
+                ) {
+                    break;
+                }
+            }
+
+            $notAnnotatedThrowables[] = $identifiedThrownThrowable;
+        }
+
+        return false === empty($notAnnotatedThrowables);
+    }
+
+    private function identifyThrownThrowables(Throw_ $node):array
+    {
+        switch (get_class($node->expr)) {
+            case New_::class:
+                return [$this->buildFQN($node)];
+            case StaticCall::class:
+                return $this->identifyThrownThrowablesInStaticCall($node);
+            default:
+                dumpf(null, $node);
+                throw new ShouldNotHappenException(sprintf('The \Throwable %s is not supported. Please, open an issue.', get_class($node->expr)));
+        }
+    }
+
+    private function identifyThrownThrowablesInStaticCall(Throw_ $node):array
+    {
+        ddf(null, 'Nodo' , $node);
+    }
```

Comment:
@TomasVotruba , Any ideas about the path to follow to access the dockblock and return type hint of the static method?

Failing test case: https://github.com/rectorphp/rector/pull/2833/files#diff-d1249893d1b00c7b6078b3d7924b48d4

Reply:
I don't understand what is needed.

Could you provide minimal PHP code snippet that explain what should return when?